### PR TITLE
fix bug in clustering

### PR DESCRIPTION
**Pull request by nilsbeck**
# Description

Fixes a bug where cluster size is 0 stops if input has 1 stop and 2 vehicles.

### CODE REVIEW

**Review by larsbeck — DISMISSED** [review]

File: templates/cluster-tsp/main.go
Comment (dirkschumacher) [inline templates/cluster-tsp/main.go]: What is the logic behind `len(input.Stops) / len(input.Vehicles)`? Why does the average number of stops per vehicle equals the maximum points?

I am asking because I am wondering if `s/v + 1` is right here or rather `ceil(float64(s)/float64(v))`?

**Review by dirkschumacher — COMMENTED** [review]
A comment

File: templates/cluster-tsp/main.go
Comment (nilsbeck) [reply templates/cluster-tsp/main.go]: no real requirement. it is mainly to avoid an input problem that could happen in the very rare case of 1s and 2v. We can also use ceil to avoid this problem, I have no opinion.

Comment (dirkschumacher) [reply templates/cluster-tsp/main.go]: Ah ok, `maximumPoints` is the upper bound on points per cluster. You could also do `(s+v - 1) / v` which is equivalent to `ceil(float64(s)/float64(v))` for integer arithmetic. However you need to ensure that `s >= 0 && v > 0` (was the case before, otherwise there is a panic). In theory `s + v` could overflow, but I think the risk is extremely low here.

Also we can move the computation outside of the loop.

Comment (dirkschumacher) [reply templates/cluster-tsp/main.go]: That being said, for `v > 0`, `math.Ceil(float64(len(...))/float64(len(...)))` may be more readable.

Comment (merschformann) [reply templates/cluster-tsp/main.go]: Did you reach a consensus here, or is another opinion required? :sweat_smile:

Comment (merschformann) [reply templates/cluster-tsp/main.go]: I think I lean towards `Ceil` too, but have no strong opinion. Will rubber stamp to potentially unblock.

**Review by merschformann — DISMISSED** [review]
See comment. :relaxed:

**Review by merschformann — APPROVED** [review]

**Review by dirkschumacher — APPROVED** [review]

**Pull request merged**
